### ISSUE TITLE
Cannot connect to irc.libera.chat using SASL

### ISSUE THREAD

**Issue by melloc** [matrix-org/matrix-appservice-irc#1353]
**Describe the bug**

I run my own synapse and matrix-appservice-irc instances, and recently tried to set things up for libera.chat. I pulled in the latest changes, set `sasl: true` in the `config.yaml` for the domain, and ran `!username` and `!storepass` for my account. However, I still see these errors in the logs:

 ```
2021-05-28 16:51:49 ERROR:client-connection melloc@irc.libera.chat: {"args":["Closing Link: example.com (SASL access only)"],"commandType":"normal","command":"ERROR","rawCommand":"ERROR"}
2021-05-28 16:51:49 INFO:client-connection disconnect()ing melloc@irc.libera.chat - raw_error
2021-05-28 16:51:49 ERROR:client-connection ConnectionInstance.connect failed after 27 attempts (raw_error)
2021-05-28 16:51:49 INFO:client-connection Retrying connection for melloc on irc.libera.chat in 15366.487665734823ms (attempts 27)
```

I'm not sure whether I'm doing something obviously wrong here or not. Here's the libera.chat portion of the `config.yaml`:

```yaml
    irc.libera.chat:
      name: "Libera"
      networkId: "liberachat"
      port: 6697
      ssl: true
      sasl: true
      sslselfsign: false
      sendConnectionMessages: true
      botConfig:
        enabled: false
        nick: "MatrixBot"
        joinChannelsIfNoUsers: true

      privateMessages:
        enabled: true
        federate: false

      dynamicChannels:
        enabled: true
        createAlias: true
        published: false
        joinRule: public
        federate: false
        aliasTemplate: "#liberachat_$CHANNEL"

      membershipLists:
        enabled: true
        floodDelayMs: 10000
        global:
          ircToMatrix:
            initial: false
            incremental: true
          matrixToIrc:
            initial: true
            incremental: true
        channels:
            ...snipped

      matrixClients:
        userTemplate: "@liberachat_$NICK"
        displayName: "$NICK"

      ircClients:
        nickTemplate: "$DISPLAY"
        allowNickChanges: true
        maxClients: 40
        idleTimeout: 0
        reconnectIntervalMs: 5000
        lineLimit: 5
```

**To Reproduce**
1. Pull latest commit from repo, and update with `npm install`
2. Set up `irc.libera.chat` with `sasl: true`
3. Try and connect with registered nickname using `!username`, `!storepass`, and `!reconnect`

**Expected behavior**

I would expect the appservice to either successfully connect using SASL or get a different different error than `SASL access only`.

**Comment by Half-Shot** [comment]
@melloc What version of the bridge are you using. I believe only develop supports a working version of SASL.

**Comment by melloc** [comment]
@Half-Shot I'm running from a clone of the `develop` branch (my current `HEAD` is 2c733393ef0fd364c2decf7ca00e38220d4b17a4) that I've run `npm install` in.

**Comment by melloc** [comment]
I modified the source to pass `debug: server.domain === 'irc.libera.chat'` to the IRC `Client`, so that I could see what was being passed to `_send`:

```
31 May 17:30:21 - Sending irc NICK/USER
31 May 17:30:21 - SEND: NICK melloc
31 May 17:30:21 - SEND: USER melloc 8 * @cody:timecube.club
31 May 17:30:21 - SEND: CAP LS 302
31 May 17:30:21 - GOT NOTICE from the server: "*** Notice -- You need to identify via SASL to use this server"
31 May 17:30:21 - Unhandled message: {
  args: [ 'Closing Link: timecube.club (SASL access only)' ],
  commandType: 'normal',
  command: 'ERROR',
  rawCommand: 'ERROR'
}
2021-05-31 17:30:21 ERROR:client-connection melloc@irc.libera.chat: {"args":["Closing Link: timecube.club (SASL access only)"],"commandType":"normal","command":"ERROR","rawCommand":"ERROR"}
2021-05-31 17:30:21 INFO:client-connection disconnect()ing melloc@irc.libera.chat - raw_error
31 May 17:30:21 - SEND: QUIT :node-irc says goodbye
31 May 17:30:21 - Connection got "end" event
2021-05-31 17:30:21 ERROR:client-connection ConnectionInstance.connect failed after 11 attempts (raw_error)
2021-05-31 17:30:21 INFO:client-connection Retrying connection for melloc on irc.libera.chat in 4691.1904349578135ms (attempts 11)
31 May 17:30:21 - Connection got "close" event
```

Taking a look at the examples for [the SASL extension](https://ircv3.net/specs/extensions/sasl-3.2), it looks like the `CAP LS` should come first. After updating `matrix-org-irc` to do this first, I am able to connect and authenticate to `irc.libera.chat`. I'll submit a PR to matrix-org/node-irc with the change.

**Comment by jaller94** [comment]
Presumably fixed by #67.

@melloc Please re-open, if the issue has not been fixed.
